java/query/query-core/src/main/java/sleeper/query/model/Query.java

```diff
@@ -169,6 +169,9 @@ public Builder processingConfig(QueryProcessingConfig processingConfig) {
         }
 
         public Query build() {
+            queryId = validateNonNull(queryId, this, "queryId field must be provided");
+            tableName = validateNonNull(tableName, this, "tableName field must be provided");
+            regions = validateNonNull(regions, this, "regions field must be provided");
```

Comment:
Please can this method be moved to the top of the builder class, just below the constructor, so we can see what fields there are to be validated?